File: Chess.Web/Chess/src/app/service/chess.service.ts
import { Injectable } from "@angular/core";
import { HttpClient } from "@angular/common/http";
import { MovementState } from "./movement-state";
import { Observable } from "rxjs";
import { MovementResult } from "./movement-result";

@Injectable({
    providedIn: "root"
})
export class ChessService {
    private apiDomain: string = "http://localhost:5001"

    constructor(private client: HttpClient) {
    }

    public move(movementState: MovementState): Observable<MovementResult> {
        return this.client.post<MovementResult>(`${this.apiDomain}/api/chess/move`, movementState);
    }
}

File: Chess.Gateways/ChessService.cs
using Chess.Gateways.Models;
using Chess.Gateways.Settings;
using Chess.Gateways.Utils;

namespace Chess.Gateways
{
    public class ChessService : IChessService
    {
        private readonly IJsonClient client;
        private readonly ChessEndpoints endpoints;

        public ChessService(IJsonClient client, ChessEndpoints endpoints)
        {
            this.client = client;
            this.endpoints = endpoints;
        }

        public void Game()
        {
        }

        public MovementResult Move(MovementState movementState)
        {
            return new MovementResult
            {
                Result = "No movement result yet!"
            };
        }
    }
}
